### PR TITLE
SmtpConnection: fix Socket not getting Disposed when LingerState cannot be set.

### PR DESCRIPTION
**Pull request by tmds**
Fixes https://github.com/dotnet/corefx/issues/40711

cc @stephentoub

### CODE REVIEW

**Comment by davidsh** [comment]
We're currently not accepting any PRs on this corefx repo. We'll be publishing the new dotnet/runtime repo in the next several days.

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs [hunk 1]
```diff
@@ -172,7 +167,13 @@ internal void Abort()
                         // interpreted correctly by the server if it's in the middle of a
                         // DATA command or some similar situation.  This may send a RST
                         // but this is ok in this situation.  Do not reuse this connection
-                        _tcpClient.LingerState = new LingerOption(true, 0);
+                        try
+                        {
+                            _tcpClient.LingerState = new LingerOption(true, 0);
+                        }
+                        catch
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs:174]: Can we catch the exact problematic exception type?

Comment (tmds) [reply src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs:174]: The dump indicates that we don't get past this line.
But we don't know what exact exception gets thrown.

**Comment by tmds** [comment]
Replaced by https://github.com/dotnet/runtime/pull/428.

**Pull request closed without merging**